localstack-core/localstack/services/kms/provider.py

```diff
@@ -992,7 +1000,10 @@ def decrypt(
         self._validate_key_state_not_pending_import(key)
 
         try:
-            plaintext = key.decrypt(ciphertext, encryption_context)
+            if not ciphertext:
+                plaintext = key.decrypt_rsa(ciphertext_blob)
```

Comment:
This implementation is very specific to RSA asymmetric keys decrypt online and encrypt offline. It would be great if we can verify that only if the key is an RSA key then it should be passed to the `decrypt_rsa` function. Also, we can explicitly mention this in a short comment above here.

Reply:
Makes sense - let me know what you think of the additional check + comments

Reply:
I would suggest to use the RSA values from `KeySpec` rather than hardcoding it here. Also to keep the code really clean consider creating a helper function, so we can do something like this: 

```
def _validate_rsa_key(self, key_spec: KeySpec) -> bool: 
        # perform the logic here and return the boolean value 
```

For the comment, we can simplify it further: 

```
# TODO: Extend the implementation to handle additional encryption/decryption scenarios 
# beyond the current support for offline encryption and online decryption using RSA keys if key id exists in 
# parameters, where `ciphertext_blob` will not be deserializable.
```

Reply:
@sannya-singal - I implemented the helper method the way I think you have in mind. I didn't see the `KeySpec` enum earlier, so thanks for pointing that out. 👍 FWIW `key.crypto_key.key_spec` is a `str`